modules/instrument_list/php/instrument_list_controlpanel.class.inc

```diff
@@ -280,24 +280,22 @@ class Instrument_List_ControlPanel extends \TimePoint
 
         // set the BVLQC flag
         $user =& \User::singleton();
-        if (isset($_REQUEST['setBVLQCStatus'])
+        if (!empty($_REQUEST['setBVLQCStatus'])
             && $user->hasPermission('bvl_feedback')
         ) {
-            $status = \Utility::nullifyEmpty(
-                $_REQUEST['setBVLQCStatus'],
-                'BVLQCStatus'
-            );
-            $this->setData('BVLQCStatus', $status);
+            $this->setData('BVLQCStatus', $_REQUEST['setBVLQCStatus']);
             return;
         }
-        if (isset($_REQUEST['setBVLQCType']) && !empty($_REQUEST['setBVLQCType'])
+        if (empty($_REQUEST['setBVLQCStatus'])
             && $user->hasPermission('bvl_feedback')
         ) {
-            $QCtype = \Utility::nullifyEmpty(
-                $_REQUEST['setBVLQCType'],
-                'BVLQCType'
-            );
-            $this->setData('BVLQCType', $QCtype);
+            $this->setData('BVLQCStatus', null);
+            return;
+        }
+        if (!empty($_REQUEST['setBVLQCType'])
```

Comment:
could these 4 if statements be condensed into 2 if/else blocks?

Reply:
@ZainVirani  the first if statement needs to check "$user->hasPermission('bvl_feedback')". 
the second if statement does not need to check this permission.

Reply:
@zaliqarosli  you are right!